source/extensions/filters/http/common/aws/credentials_provider.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include <string>
+
+#include "envoy/common/pure.h"
+
+#include "absl/types/optional.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Common {
+namespace Aws {
+
+class Credentials {
+public:
+  Credentials() = default;
+
+  ~Credentials() = default;
+
+  Credentials(const std::string& access_key_id, const std::string& secret_access_key)
```

Comment:
Nit: we tend to use `absl::string_view` rather than `const std::string&` in new code, as it's strictly more general. You can still turn this into a string copy below.

Reply:
clang-tidy will suggest pass by value here and std::move below for constructors like this.
https://clang.llvm.org/extra/clang-tidy/checks/modernize-pass-by-value.html

By doing this you _may_ save one copy if caller does move too.